### PR TITLE
AIP-84 List Task Instances

### LINKED ISSUE #43748: AIP-84 Get Task Instances
### Description

Currently the Get Task Instances public endpoint is at `api_connexion/endpoints/task_instance_endpoint.py` under `get_task_instances`. We need to migrate it to the `api_fastapi/views/public/task_instances.py` under a `get_task_instances` or similar.

Features and functionality of the endpoint will remain unchanged.

### Use case/motivation

AIP-84 cWiki Link: https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-84+UI+REST+API

### Related issues

Related to: https://github.com/apache/airflow/issues/42370

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by pierrejeambrun**
related to https://github.com/apache/airflow/issues/42370

On top of https://github.com/apache/airflow/pull/43642, only last commit is relevant.

Closes: https://github.com/apache/airflow/issues/43748

### CODE REVIEW

File: airflow/api_fastapi/core_api/openapi/v1-generated.yaml [hunk 1]
```diff
@@ -2470,6 +2470,198 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/HTTPValidationError'
+  /public/dags/{dag_id}/dagRuns/{dag_run_id}/taskInstances/:
```
Comment (bbovenzi) [inline airflow/api_fastapi/core_api/openapi/v1-generated.yaml:2473]: Will we support wildcard ids to list all task instances for all runs or dags? Can be another PR

File: airflow/api_fastapi/core_api/openapi/v1-generated.yaml [hunk 2]
```diff
@@ -2470,6 +2470,198 @@ paths:
...
+        schema:
+          type: string
+          title: Dag Id
+      - name: dag_run_id
+        in: path
+        required: true
+        schema:
+          type: string
+          title: Dag Run Id
+      - name: execution_date_gte
```
Comment (bbovenzi) [inline airflow/api_fastapi/core_api/openapi/v1-generated.yaml]: execution_date is deprecated and was renamed to logical_date

File: airflow/api_fastapi/core_api/routes/public/task_instances.py [hunk 3]
```diff
@@ -198,3 +198,75 @@ async def get_mapped_task_instance(
...
+    pool: QueryTIPoolFilter,
+    queue: QueryTIQueueFilter,
+    executor: QueryTIExecutorFilter,
+    limit: QueryLimit,
+    offset: QueryOffset,
+    order_by: Annotated[
+        SortParam,
+        Depends(
+            SortParam(
+                ["id", "state", "duration", "start_date", "end_date", "map_index"],
```
Comment (bbovenzi) [inline airflow/api_fastapi/core_api/routes/public/task_instances.py:229]: I could see sort by try_number, logical_date, data_interval_start/end, rendered_map_index  would also be useful. Can be another PR.

File: airflow/api_fastapi/core_api/openapi/v1-generated.yaml (hunk 1 above)
Comment (pierrejeambrun) [reply airflow/api_fastapi/core_api/openapi/v1-generated.yaml:2473]: Yes this is currently supported and tested.

File: airflow/api_fastapi/core_api/openapi/v1-generated.yaml (hunk 2 above)
Comment (pierrejeambrun) [reply airflow/api_fastapi/core_api/openapi/v1-generated.yaml]: Done, thanks.

File: airflow/api_fastapi/core_api/openapi/v1-generated.yaml (hunk 1 above)
Comment (pierrejeambrun) [reply airflow/api_fastapi/core_api/openapi/v1-generated.yaml:2473]: Manually added a piece of documentation for that.

Screenshot 2024-11-07 at 10 48 41

File: airflow/api_fastapi/core_api/routes/public/task_instances.py (hunk 3 above)
Comment (pierrejeambrun) [reply airflow/api_fastapi/core_api/routes/public/task_instances.py:229]: Issue created [here](https://github.com/apache/airflow/issues/43778), I'm assigned, and can do that when migration stuff are done, I don't want to be too late for migrating the API.

Also there are some changes to bring to the common `SortParam` to support aliasing for `execution_date/logical_date`

**Review by bbovenzi — APPROVED** [review]
